libr/cons/dietline.c

```diff
@@ -218,12 +217,18 @@ static int r_line_readchar_win(int *vch) { // this function handle the input in
 	ret = ReadConsoleInput (h, &irInBuf, 1, &out);
 	r_cons_sleep_end (bed);
 	if (ret < 1) {
+		free (buf);
 		return 0;
 	}
 	if (irInBuf.EventType == KEY_EVENT) {
 		if (irInBuf.Event.KeyEvent.bKeyDown) {
-			if (irInBuf.Event.KeyEvent.uChar.AsciiChar) {
-				*buf = irInBuf.Event.KeyEvent.uChar.AsciiChar;
+			if (irInBuf.Event.KeyEvent.uChar.UnicodeChar) {
+				free (buf);
+#if UNICODE
```

Comment:
I suggest to use function pointer here and set the proper function from the place where UNICODE/ASCII was determined.

Reply:
reworked the r_sys_conv macros to do this